modules/hcp-agentserviceconfig.adoc

```diff
@@ -0,0 +1,116 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="hcp-agentserviceconfig_{context}"]
+= Deploying AgentServiceConfig resources
+
+The `AgentServiceConfig` custom resource is an essential component of the Assisted Service add-on that is part of {mce-short}. It is responsible for bare metal cluster deployment. When the add-on is enabled, you deploy the `AgentServiceConfig` resource to configure the add-on.
+
+In addition to configuring the `AgentServiceConfig` resource, you need to include additional config maps to ensure that {mce-short} functions properly in a disconnected environment.
+
+. Configure the custom registries by adding the following config map, which contains the disconnected details to customize the deployment:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: custom-registries
+  namespace: multicluster-engine
+  labels:
+    app: assisted-service
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    -----END CERTIFICATE-----
+  registries.conf: |
+    unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]
+
+    [[registry]]
+    prefix = ""
+    location = "registry.redhat.io/openshift4"
+    mirror-by-digest-only = true
+
+    [[registry.mirror]]
+      location = "registry.dns.base.domain.name:5000/openshift4" <1>
+
+    [[registry]]
+    prefix = ""
+    location = "registry.redhat.io/rhacm2"
+    mirror-by-digest-only = true
+    ...
+    ...
+----
++
+<1> Replace `dns.base.domain.name` with the DNS base domain name.
++
+The object contains two fields:
+
+* Custom CAs: This field contains the Certificate Authorities (CAs) that are loaded into the various processes of the deployment.
+* Registries: The `Registries.conf` field contains information about images and namespaces that need to be consumed from a mirror registry rather than the original source registry.
+
+. Configure the Assisted Service by adding the `AssistedServiceConfig` object, as shown in the following example:
++
+[source,yaml]
+----
+---
+apiVersion: agent-install.openshift.io/v1beta1
+kind: AgentServiceConfig
+metadata:
+  annotations:
+    unsupported.agent-install.openshift.io/assisted-service-configmap: assisted-service-config <1>
+  name: agent
+  namespace: multicluster-engine
+spec:
+  mirrorRegistryRef:
+    name: custom-registries <2>
+  databaseStorage:
+    storageClassName: lvms-vg1
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 10Gi
+  filesystemStorage:
+    storageClassName: lvms-vg1
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 20Gi
+  osImages: <3>
+  - cpuArchitecture: x86_64
+    openshiftVersion: "4.14"
+    rootFSUrl: http://registry.dns.base.domain.name:8080/images/rhcos-414.92.202308281054-0-live-rootfs.x86_64.img <4>
+    url: http://registry.dns.base.domain.name:8080/images/rhcos-414.92.202308281054-0-live.x86_64.iso
+    version: 414.92.202308281054-0
```

Comment:
Maybe we can add another sample here to remark that we need to add every OCP release we wanna deploy as HCP by adding that release to this file. Sample:

```
- cpuArchitecture: x86_64
   openshiftVersion: "4.15"
   rootFSUrl: http://registry.dns.base.domain.name:8080/images/rhcos-415.92.202403270524-0-live-rootfs.x86_64.img
   url: http://registry.dns.base.domain.name:8080/images/rhcos-415.92.202403270524-0-live.x86_64.iso
   version: 415.92.202403270524-0
```

Reply:
Added